Settings: Add Tracks events for various site/SEO changes

jeremeylduvall opened the pull request:
This PR adds some additional Tracks events to monitor changes in the following items:
- Updated site title/tagline
- Changed site privacy
- Added site verification services under Traffic
- Deactivated/Activated Google AMP
- Added Google Analytics

This is part of some more in-depth digging @chrisfromthelc is doing along the lines of SEO and the Business plan.

### To test
1. Load up this branch. Use `localStorage.setItem('debug', 'calypso:analytics*');` to log out Tracks events to your console.
2. First, visit http://calypso.localhost:3000/settings/general/. Change the site title and tagline. Press "Save Settings." You should see the `calypso_settings_site_tagline/title_updated` event get recorded.
3. Next, change the site privacy on the same page and press "Save Settings". The `calypso_settings_site_privacy_updated` should get recorded with the privacy passed in as a prop.
4. Try adding some of the verification services under http://calypso.localhost:3000/settings/traffic/. The `calypso_seo_tools_site_verification_updated` event should fire with a prop of the verified service. I've pasted example codes to use at the bottom here for ease of reference.
5. Try deactivating/activating AMP on a WP.com site on /settings/traffic. `calypso_seo_settings_amp_updated` should get fired with a prop of whether AMP is enabled or not.
6. Fill out the Google Analytics field (example value below) and save the settings. `calypso_seo_settings_google_analytics_updated` should fire.

**Example verification codes**
These are the examples we provide to use for testing.
- Google: `<meta name="google-site-verification" content="f0w4NjmVso-ol9myrC9J4mbOnsZm_dj_rBK3VoOjNRo" />`
- Bing: `<meta name="msvalidate.01" content="1234" />`
- Pinterest: `<meta name="p:domain_verify" content="1234" />`
- Yandex: `<meta name="yandex-verification" content="1234" />`
- Google Analytics: UA-90363941-1

tyxla commented on client/my-sites/site-settings/amp/wpcom.jsx:
```diff
@@ -104,5 +106,11 @@ class AmpWpcom extends Component {
 		siteSlug: getSelectedSiteSlug( state ),
-	} )
+	} ),
+	( dispatch ) => {
```
There is a short syntax that automatically binds the action creators to `dispatch`, so

```
( dispatch ) => {
	const trackTracksEvent = ( name, props ) => dispatch( recordTracksEvent( name, props ) );
	return {
		trackTracksEvent,
	};
},
```

could be shortened to just:

```
{ recordTracksEvent }
```

Then we could also call it like this: `this.props.recordTracksEvent()` and avoid the tongue-twister prop name 😉

tyxla commented on client/my-sites/site-settings/wrap-settings-form.jsx:
```diff
@@ -100,10 +100,29 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 				event.preventDefault();
 			}
 
+			forEach( dirtyFields, function( value ) {
```
Any good reason to use `forEach` here? I think it's generally slower, how about using `map` or something similar here?

tyxla commented on client/my-sites/site-settings/amp/wpcom.jsx:
```diff
@@ -32,7 +33,8 @@ class AmpWpcom extends Component {
 	handleToggle = () => {
-		const { fields, submitForm, trackEvent, updateFields } = this.props;
+		const { fields, submitForm, trackEvent, updateFields, trackTracksEvent } = this.props;
+		trackTracksEvent( 'calypso_seo_settings_amp_updated', { amp_is_enabled: ! fields.amp_is_enabled } );
```
I'd store `{ amp_is_enabled: ! fields.amp_is_enabled }` in a variable above, then use it here and in the `updateFIelds` call for 2 reasons:

* To avoid repetition (stay DRY)
* To reduce the possibility for errors if we decide to change this and change only one of the occurrences.

jeremeylduvall commented:
Thanks! I've updated the PR as requested to use a variable for `amp_is_enabled` and `map` instead of `forEach`. With the `mapDispatchToProps`, I originally had that implementation, but I was running into some linting errors. Thanks to [this](https://stackoverflow.com/questions/37682705/avoid-no-shadow-eslint-error-with-mapdispatchtoprops), I realize it was because of the destructuring. Fixed up now. Will wait for CircleCI to pass and probably ship this tonight/tomorrow morning.
